src/App.tsx

```diff
@@ -15,48 +17,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum TypeOfSort {
+  name = 'name',
+  length = 'length',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sort: string,
+  reverse: boolean,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sort) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sort) {
+        case TypeOfSort.name:
+          return good1.localeCompare(good2);
+
+        case TypeOfSort.length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState('');
+  const [reverse, setReverse] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortType, reverse);
+
+  const reset = () => {
+    setSortType('');
+    setReverse(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn(
+            'button is-info',
+            { 'is-light': sortType !== TypeOfSort.name },
+          )}
+          onClick={() => setSortType(TypeOfSort.name)}
```

Comment:
consider to make one hanler factory for all sort fields which will take sortType and return arrow function.
```
const handleSort = (sortType: TypeOfSort) => () => setSortType(sortType);
```
```suggestion
          onClick={handleSort(TypeOfSort.NAME)}
```